### ISSUE TITLE
Img Writer: Writing Tiffs is _very_ slow

### ISSUE THREAD

**Issue by dietzc** [knime-ip/knip#131]
Writing a tiff with e.g. 1024,1024 and Z=51, Channel  = 3 in this dimension order is super slow. One reason might be, that it seems that nearly every pixel write action is logged somewhere. @gab1one please reproduce and double check!

**Comment by gab1one** [comment]
Yes, the tiff writers (tiff and ome.tiff) are about 6.000% slower than eq. formats like ics. Even with small images. 
One example:  
- SCIFIO can write the same image (128,128, C=1  Z=1, T=50,   )  `137.165 ±  2.115` times a second as ICS file but only about `2.093 ±  0.063` a second as (ome.)tiff file.    
take a look at the [benchmark results](https://gist.github.com/gab1one/d359578f0686aa609055)

I removed logging  from  the TIFF format to check your hypothesis but it did not produce any meaningful performance improvement (up to 0.4 more  operations more per second!). 
[without logging](https://gist.github.com/gab1one/029928b79a2aec1a7f1e)

Larger files take much longer to write but show the same effect.

**Comment by dietzc** [comment]
The slow down seems to be related To file size. Can you test with a 100mb image with z channrl time and different permutations?

**Comment by gab1one** [comment]
see  https://github.com/knime-ip/knip/commit/46b6799b83512270b9752ec555cef5cbdf95528b for the first big improvement. Writing the 160mb file sequentially now takes about 10 seconds instead of the 44 it took before

**Comment by dietzc** [comment]
great work @gab1one. Reading & Writing Tiffs (and also other formats) is very often used by the users and therefore performance is crucial.

**Comment by gab1one** [comment]
Further improvements depend on changes in SCIFIO, I will talk with Mark about that.

**Comment by imagejan** [comment]
Any news on this? I'm still suffering severe performance issues with the Image Writer.

Here's an example benchmark workflow:
[Image Writer Performance.zip](https://github.com/knime-ip/knip/files/375462/Image.Writer.Performance.zip)

I get the following writing times for the different file formats:
```
Node 2    tif       254156 

Node 3    ome.tif   191286 

Node 4    ics       549 
```

**Comment by dietzc** [comment]
I think this is still a SCIFIO Problem. At the moment I can only recommend to use `ics` in favour of tiff. @ctrueden what do you think?

**Comment by ctrueden** [comment]
I am aware of a SCIFIO issue for improving the performance of TIFF _reading_: scifio/scifio#7.

But is there an issue for improving the behavior of TIFF _writing_? I couldn't find it.

Has anyone profiled why TIFF writing is slow? I can't imagine it would be that hard to fix. #famouslastwords

**Comment by dietzc** [comment]
@gab1one can you reproduce the problem and open an issue in scifio?

**Comment by imagejan** [comment]
The slowness issue is reproducible in ImageJ, the relevant SCIFIO issue is https://github.com/scifio/scifio/issues/310.

**Comment by dietzc** [comment]
see also: https://tech.knime.org/forum/knime-image-processing/image-writer-cannot-write-to-network-drives#comment-48710

**Comment by tibuch** [comment]
Any news regarding tif writing speed? I am currently using the image writer to move my data to the python-nodes and it takes a very long time.

**Comment by ctrueden** [comment]
@tibuch Sorry for the long delay in reply. Unfortunately, no one is working on TIFF performance. I am very sorry this has sat for so long, but the fact is that we still do not have a dedicated SCIFIO developer who has time to prioritize this issue... yet.
